### PR TITLE
Initial docker setup

### PR DESCRIPTION
**Pull request by VickTor61**

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Include any files or directories that you don't want to be copied to your
+# container here (e.g., local build artifacts, temporary files, etc.).
+#
+# For more help, visit the .dockerignore file reference guide at
+# https://docs.docker.com/engine/reference/builder/#dockerignore-file
+
+**/.DS_Store
```
[the remark this answers is not in the document]
Comment (VickTor61) [reply .dockerignore:7]: Yeah these were created when i intialised docker. Let me clean it up

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+FROM node:21-bullseye
+
+# Set the working directory inside the container
+WORKDIR /app
+
+# Copy the entire project
+COPY . .
+
+# Copy the package.json and yarn.lock files
+COPY package.json yarn.lock ./
```
Comment (mrdjohnson) [inline Dockerfile:11]: we already copied the entire project, do we still need this as well?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+# Set the yarn version to be that defined as packageManager
+RUN yarn set version 4.1.1
+
+# Install the dependencies
+RUN yarn install --frozen-lockfile
```
Comment (mrdjohnson) [inline Dockerfile:17]: not important: I ran `docker build` with this flag: `--progress=plain` to see the logs, it showed this:

```
 The --frozen-lockfile option is deprecated; use --immutable and/or --immutable-cache instead
```

maybe we can use immutable instead?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+# Build the Vite app with the base path
+RUN yarn run build 
+
+# Expose the port on which the app will run 
+EXPOSE 5173
+
+# Start the app
+CMD ["yarn", "preview", "--host", "--port", "5173"]
```
Comment (mrdjohnson) [inline Dockerfile:26]: 👏  nice

File: compose.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+services:
+  app:
+    build:
+      context: .
+      target: final
```
Comment (mrdjohnson) [inline compose.yaml:5]: what command are you running to get docker compose to work? 

`docker compose up` says it cannot find target final

**Review by mrdjohnson — CHANGES_REQUESTED** [review]
docker compose part does not seem to be working?

**Comment by VickTor61** [comment]
> docker compose part does not seem to be working?

Yeah i mentioned i just added the boiler plate. Needed to know if we needed it at the moment

File: Dockerfile (hunk 3 above)
Comment (VickTor61) [reply Dockerfile:17]: I didn't know it was deprecated. Didn't see any warnings on my end but will update that.

File: Dockerfile (hunk 2 above)
Comment (VickTor61) [reply Dockerfile:11]: Naa. another oversight on my end. Already fixed that. Pushing now

**Pull request merged**